### DISCUSSION TITLE
lib9c에서 정의한 enum타입 관리 및 적극적으로 활용하기

### DISCUSSION THREAD

**Discussion by eugene-doobu** [planetarium/NineChronicles#4447] [Ideas]
160 rc를 준비하면서 장비의 grade enum타입에 접근하는 작업이 2개 있었음.

- 신규 장비등급 추가: https://github.com/planetarium/lib9c/commit/089b0c00ed02b75001d0ee280f46acf2ffdd8dff
- 장비등급별 아이템 필터 추가: https://github.com/planetarium/NineChronicles/pull/4413

하지만 Grade와 같은 enum타입을 프로젝트 어디에서도 사용하고 있지 않음.
-> 보통 클라이언트 내부에서 데이터를 처리할때에는 int와 같은 정수 타입으로 적용
-> 클라이언트 인벤토리에서는 자신만의 enum타입을 추가해서 사용
-> 아이템 필터에 적용해볼까 했는데, 현재 해당 기능 구현을 flag enum을 기반으로 하여 당장 해당 enum타입을 적용하는데 애매한 면이 있었음

총 같은 역할을 하는 enum타입이 3개가 있는 상태. 굳이 enum을 새로 파서 사용하는거면 그냥 lib9c에 있는 enum타입만을 사용해서 기능을 구현하는 것이 좋음

아이템 필터 기능은 현재 필터 옵션에서 필요한 enum타입들을 확인한 후, 클라이언트 쪽에서는 정수타입 변수에 bit flag를 이용하여 기능을 재구현할 계획을 가지고 있음. 다른 콘텐츠에서도 확인 후 가능하면 lib9c에 정의된 enum타입을 사용하는것이 좋아보임.

**Comment by ipdae** [comment]
체인(액션실행), 클라이언트가 공용으로 사용되는 부분은 구현을 lib9c쪽에 해두고 양쪽에서 사용하는게 원래 의도긴했습니다.
클라이언트에서만 필요한 부분들이 생기면서 중복구현이 된것같은데 정리가 가능하다면 좋을것같습니다.

**Comment by jonny-jeahyunchoi** [comment]
비슷하게 Elemental, ElementalType도 있습니다~!
클라이언트에서  뭔가 lib9 enum들을 확장해서 사용하는부분들이 더있는것같은데 일단 발견된건 저정도네요
